packages/expo-video/android/src/main/java/expo/modules/video/VideoView.kt

```diff
@@ -195,6 +225,45 @@ open class VideoView(context: Context, appContext: AppContext, useTextureView: B
   }
 
   fun enterFullscreen() {
+    if(isInFullscreen){ return }
+    Log.d(LOG_TAG, "enterFullscreen")
+
+    // Set before starting to avoid entering PiP unintentionally
+    isInFullscreen = true
+
+    if(useFullscreenDialog){ enterFullscreenDialog() }
+    else { enterFullscreenActivity() }
+
+    onFullscreenEnter(Unit)
+    applyPiPParams(currentActivity, false, calculateCurrentPipAspectRatio())
+  }
+
+  /**
+   * NOTE: This method should only be called by [enterFullscreen].
+   */
+  private fun enterFullscreenDialog(){
+    fullscreenPlayerDialog = FullscreenPlayerDialog(context, playerView, ::exitFullscreen)
+    fullscreenPlayerDialog?.show()
+  }
+
+  /**
+   * NOTE: This method should only be called by [exitFullscreen].
+   */
+  private fun exitFullscreenDialog() {
+    val dialog = fullscreenPlayerDialog ?: return
+    dialog.dismiss()
+
+    // Move playerView back to normal view
+    val playerViewParent = (playerView.parent as? ViewGroup)
+    playerViewParent?.removeView(playerView)
+    addView(playerView, FrameLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT))
```

Comment:
Should these layout params be hardcoded like this? Could that cause any issues if the original `playerView` was using a different configuration?